doc/index.md

```diff
@@ -236,114 +292,131 @@ local myconnection = http_client.connect {
 
 
-### `h1_connection:peername()` <!-- --> {#http.h1_connection:peername}
+### `h1_connection.version` <!-- --> {#http.h1_connection.version}
 
-See [`connection:peername()`](#connection:peername)
+Used to specificy the http version used for the connection handshake. Valid values are:
+- `1.0` 
+- `1.1`
 
 
 ### `h1_connection:clearerr(...)` <!-- --> {#http.h1_connection:clearerr}
 
+ 
 
 ### `h1_connection:take_socket()` <!-- --> {#http.h1_connection:take_socket}
 
 
-### `h1_connection:shutdown(dir)` <!-- --> {#http.h1_connection:shutdown}
-
-See [`connection:shutdown()`](#connection:shutdown)
-
-
-### `h1_connection:close()` <!-- --> {#http.h1_connection:close}
-
-See [`connection:close()`](#connection:close)
-
-
-### `h1_connection:new_stream()` <!-- --> {#http.h1_connection:new_stream}
-
-See [`connection:new_stream()`](#connection:new_stream)
-
-
-### `h1_connection:get_next_incoming_stream(timeout)` <!-- --> {#http.h1_connection:get_next_incoming_stream}
-
-See [`connection:get_next_incoming_stream()`](#connection:get_next_incoming_stream)
-
-
 ### `h1_connection:flush(...)` <!-- --> {#http.h1_connection:flush}
 
+Flushes all buffered data on the socket. Returns true on success. Returns false and the error if the socket fails to flush.
+
 
 ### `h1_connection:read_request_line(timeout)` <!-- --> {#http.h1_connection:read_request_line}
 
+Gets the request method, requested path and HTTP version for an incoming request. On error, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_status_line(timeout)` <!-- --> {#http.h1_connection:read_status_line}
 
+Gets the HTTP version (1 or 1.1) , status code and a reason description if applicable. If the socket could not be read, returns nil, an error message and an error number.
+
 
 ### `h1_connection:read_header(timeout)` <!-- --> {#http.h1_connection:read_header}
 
+Returns the HTTP header (?key, val) from an incoming request. If the header could not be read, the function returns nil an error and an error message.
+
 
 ### `h1_connection:read_headers_done(timeout)` <!-- --> {#http.h1_connection:read_headers_done}
 
+Checks if all chunks of the HTTP headers have been fully received. Returns nil, an error and an error message if the socket cannot be read.
+
 
 ### `h1_connection:read_body_by_length(len, timeout)` <!-- --> {#http.h1_connection:read_body_by_length}
 
+Get the number of bytes from the buffered HTTP request body. Use a negative number for *up to* that number of bytes. Asserts if len is not a number.
+
 
 ### `h1_connection:read_body_till_close(timeout)` <!-- --> {#http.h1_connection:read_body_till_close}
 
+Reads the entire request body. This function will yield until complete or `timeout` is expired.
```

Comment:
Reads data until the connection is closed.
Again: please avoid using the word "request" for this function.